common/remote/sockfile.cpp

```diff
@@ -2697,13 +2697,26 @@ ISocket *checkSocketSecure(ISocket *socket)
 
     if ( (pport == securitySettings.daFileSrvSSLPort) && (!socket->isSecure()) )
     {
-        Owned<ISecureSocket> ssock;
 #ifdef _USE_OPENSSL
-        ssock.setown(createSecureSocket(LINK(socket), ClientSocket));
-        int status = ssock->secure_connect();
-        if (status < 0)
-            throw createDafsException(DAFSERR_connection_failed,"Failure to establish secure connection");
-        return ssock.getClear();
+        Owned<ISecureSocket> ssock;
+        try
+        {
+            ssock.setown(createSecureSocket(LINK(socket), ClientSocket));
+            int status = ssock->secure_connect();
+            if (status < 0)
+                throw MakeStringException(-1, "Failure to establish secure connection");
```

Comment:
It may not be possible, but now if anything other than the the line above throws an exception that is a different code, it will be lost.

It would be more correct to throw a dafs exception here as before and catch and rethrow it after clearing up.
That was if anything else throws a different exception with a different code, it won't be lost.